core/src/commonMain/kotlin/com/xebia/functional/xef/llm/MetricManagement.kt

```diff
@@ -7,23 +7,39 @@ import com.xebia.functional.xef.prompt.Prompt
 suspend fun CreateChatCompletionResponse.addMetrics(
   conversation: Conversation
 ): CreateChatCompletionResponse {
-  conversation.metric.parameter("model", model)
-  conversation.metric.parameter(
-    "tokens",
-    "${usage?.promptTokens} (prompt) + ${usage?.completionTokens} (completion) = ${usage?.totalTokens}"
-  )
+  conversation.metric.parameter("openai.chat_completion.model", model)
+  usage?.let {
+    conversation.metric.parameter("openai.chat_completion.prompt.token.count", "${it.promptTokens}")
+    conversation.metric.parameter(
+      "openai.chat_completion.completion.token.count",
+      "${it.completionTokens}"
+    )
+    conversation.metric.parameter("openai.chat_completion.token.count", "${it.totalTokens}")
+  }
   return this
 }
 
 suspend fun <T> Prompt<T>.addMetrics(conversation: Conversation) {
+  conversation.metric.parameter("openai.chat_completion.prompt.message.count", "${messages.size}")
+
+  conversation.metric.parameter(
+    "openai.chat_completion.prompt.messages_roles",
+    messages.map { it.completionRole().value }
+  )
+  conversation.metric.parameter(
+    "openai.chat_completion.prompt.last-message",
+    messages.lastOrNull()?.contentAsString() ?: "empty"
+  )
+  conversation.metric.parameter(
+    "openai.chat_completion.conversation_id",
+    conversation.conversationId?.value ?: "none"
+  )
   conversation.metric.parameter(
-    "number-of-messages",
-    "${messages.size} (${messages.map { it.completionRole().value.firstOrNull() ?: "" }.joinToString("-")})"
+    "openai.chat_completion.functions",
+    if (functions.isEmpty()) listOf("no") else functions.map { it.name }
   )
```

Comment:
**Nit**, wondering if we should rename `no` to `none` or if we should omit `"openai.chat_completion.functions"` if `functions.isEmpty()`.

So:

```
if(functions.isNotEmpty()) {
  conversation.metric.parameter(
    "openai.chat_completion.functions",
     functions.map { it.name }
  )
}
```

Reply:
I like this approach.